### PR TITLE
Fix docker run

### PR DESCRIPTION
**Pull request by iamstarkov**

### CODE REVIEW

**Comment by iamstarkov** [comment]
@paulmelnikow will you have time to review this one?

**Comment by paulmelnikow** [comment]
The doc change looks great.

We should not be running the dev server in docker. Instead we should run the dev build and then run the badge server, like we do in Heroku. I wrote at https://github.com/badges/shields/issues/1373#issuecomment-353477389 about the changes needed. Could you take a look?

**Comment by iamstarkov** [comment]
@paulmelnikow i think i nailed production related changes. pls take a lok

File: Dockerfile [hunk 1]
```diff
@@ -10,10 +10,15 @@ WORKDIR /usr/src/app
...
 COPY package.json /usr/src/app/
-RUN npm install && \
-    rm -rf /tmp/npm-* /root/.npm
+RUN npm install
 COPY . /usr/src/app
+RUN npm run build
+RUN rm -rf /tmp/npm-* /root/.npm
+RUN npm run build
+RUN npm prune --production
+RUN npm cache clean --force
```
Comment (paulmelnikow) [inline Dockerfile]: I wonder if the `cache clean` makes the `rm -rf` obsolete.

File: Dockerfile [hunk 2]
```diff
@@ -10,10 +10,15 @@ WORKDIR /usr/src/app
 ARG NODE_ENV
 ENV NODE_ENV $NODE_ENV
 COPY package.json /usr/src/app/
-RUN npm install && \
-    rm -rf /tmp/npm-* /root/.npm
+RUN npm install
 COPY . /usr/src/app
+RUN npm run build
+RUN rm -rf /tmp/npm-* /root/.npm
+RUN npm run build
```
Comment (paulmelnikow) [inline Dockerfile]: Looks like a duplicate `npm run build`.

File: Dockerfile (hunk 1 above)
Comment (iamstarkov) [reply Dockerfile]: i think so, lets remove it

Comment (iamstarkov) [reply Dockerfile]: fixed

File: Dockerfile (hunk 2 above)
Comment (iamstarkov) [reply Dockerfile]: fixed

**Comment by iamstarkov** [comment]
there is a problem with nodejs 6 in CircleCI, not sure how to fix it

**Comment by iamstarkov** [comment]
my guess node-version-check blocks the start of the service, so services tests are failing

**Comment by paulmelnikow** [comment]
The service-test harness needs the PR title, and it gets this via Shields production. But Shields production is having some intermittent issues hitting the Github API. I opened #1453 to fix this by fetching from Github directly.

**Comment by paulmelnikow** [comment]
Thanks so much for fixing these issues! 🏅

**Comment by iamstarkov** [comment]
@paulmelnikow thanks for quick reply. what do i do in the meantime?

**Comment by iamstarkov** [comment]
hurray 🎉

**Comment by iamstarkov** [comment]
@paulmelnikow i have a question. is this official docker image? https://hub.docker.com/r/beevelop/shields/

**Comment by iamstarkov** [comment]
seems like its not

**Comment by iamstarkov** [comment]
because this is the repo for it https://github.com/beevelop/docker-shields

**Comment by iamstarkov** [comment]
does this repo have configured and automated way to publish its docker image to docker hub?

**Comment by paulmelnikow** [comment]
No, though if we could add some tests for the image, I'd be happy to publish it officially.

**Comment by iamstarkov** [comment]
@paulmelnikow do you refer to integration/e2e tests?

**Comment by paulmelnikow** [comment]
I'm not sure exactly what to call them. The test bracket should include a server running inside the container, and a client outside the container.

Fetching a static badge would suffice: basically loading http://localhost/badge/foo-bar-blue.svg and http://localhost/badge/foo-bar-blue.png and making sure they're valid SVG and PNG.

**Pull request merged**